### PR TITLE
[compatibility] Add compatibility notices for pp-number

### PR DESCRIPTION
**Pull request by tkoeppe**
Addresses #680.

### CODE REVIEW

File: source/compatibility.tex [hunk 1]
```diff
@@ -853,6 +853,21 @@
...
+\rationale Necessary to enable hexadecimal floating literals.
+\effect
+Valid \CppIII code may fail to compile or produce different results in
+this International Standard. Specifically, character sequences like \tcode{0p+0}
+are three separate tokens in \CppIII, but one single token in this International
+Standard. For example:
+
+\begin{codeblock}
+#define F(a) b ## a
+int b0p = F(0p0+0);  // ill-formed; equivalent to ``\tcode{int b0p = b0p + 0;}\!'' in C++ 2003
```
Comment (timsong-cpp) [inline source/compatibility.tex]: Is `F(0p+0)` meant?

Comment (tkoeppe) [inline source/compatibility.tex]: @timsong-cpp: Yes :-)

**Comment by zygoloid** [comment]
We generally only list differences against other languages in one subclause of Annex C.

**Comment by tkoeppe** [comment]
@zygoloid: So where should this go: break from C++14, or from C++03?

**Comment by tkoeppe** [comment]
@zygoloid: Now there's only a single compatibility note against C++14, with the both examples.

Preview:
image

File: source/compatibility.tex [hunk 2]
```diff
@@ -1511,6 +1511,21 @@
...
 
+\ref{lex.ppnumber}
+\change \grammarterm{pp-number} can contain \tcode{p} \grammarterm{sign} and
+\tcode{P} \grammarterm{sign}
+\rationale Necessary to enable hexadecimal floating literals.
+\effect
+Valid \CppXI code may fail to compile or produce different results in
+this International Standard. Specifically, character sequences like \tcode{0p+0}
+and \tcode{0e1_p+0} are three separate tokens each in \CppXIV, but one single token
+in this International Standard.
```
Comment (timsong-cpp) [inline source/compatibility.tex]: "Valid \CppXI code"...."in \CppXIV"

Comment (tkoeppe) [inline source/compatibility.tex]: Well spotted, thanks! Fixed.

**Comment by tkoeppe** [comment]
@zygoloid: PTAL

**Comment by tkoeppe** [comment]
Updated, thanks @timsong-cpp. New screenshot:

image


@zygoloid: PTAL

**Pull request merged**